x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/QueryRule.java

```diff
@@ -113,14 +114,33 @@ public QueryRule(StreamInput in) throws IOException {
 
     private void validate() {
         if (type == QueryRuleType.PINNED) {
-            if (actions.containsKey("ids") == false && actions.containsKey("docs") == false) {
-                throw new ElasticsearchParseException("Pinned Query rule actions must contain either ids or docs");
+            boolean ruleContainsPinnedIds = actions.containsKey(IDS_FIELD.getPreferredName());
+            boolean ruleContainsPinnedDocs = actions.containsKey(DOCS_FIELD.getPreferredName());
+            if (ruleContainsPinnedIds && ruleContainsPinnedDocs) {
```

Comment:
we might use XOR here and have a single check (I had to actually look for the XOR operator in Java: `^` 😆 )

Reply:
Thanks, you're right that's a lot cleaner!